pkg/sippyserver/server.go

```diff
@@ -1011,12 +1015,24 @@ func (s *Server) jsonJobRunRiskAnalysis(w http.ResponseWriter, req *http.Request
 
 		jobRunTestCount = jobRun.TestCount
 
-		// We don't expect the caller to fully populate the ProwJob, just it's name,
+		// We don't expect the caller to fully populate the ProwJob, just its name;
 		// override the input by looking up the actual ProwJob so we have access to release and variants.
 		job := &models.ProwJob{}
 		res := s.db.DB.Where("name = ?", jobRun.ProwJob.Name).First(job)
 		if res.Error != nil {
-			failureResponse(w, http.StatusBadRequest, fmt.Sprintf("unable to find ProwJob: %s", jobRun.ProwJob.Name))
+			if errors.Is(res.Error, gorm.ErrRecordNotFound) {
+				// sippy does not import all jobs to its prow_jobs table; for context see
+				// https://redhat-internal.slack.com/archives/C02K89U2EV8/p1736972504210699
+				// for example, PR jobs on GA releases are excluded, and various other jobs
+				errMsg := fmt.Sprintf("ProwJob '%s' is not included in imported jobs so risk analysis will not run.", jobRun.ProwJob.Name)
+				result := apitype.ProwJobRunRiskAnalysis{
+					OverallRisk: apitype.JobFailureRisk{Level: apitype.FailureRiskLevelUnknown, Reasons: []string{errMsg}},
+				}
+				api.RespondWithJSON(http.StatusOK, w, result)
+			} else {
+				failureResponse(w, http.StatusBadRequest, fmt.Sprintf("unable to find ProwJob '%s': %v", jobRun.ProwJob.Name, res.Error))
```

Comment:
Nit really but, is that the right message?  Seems like ErrRecordNotFound is the 'unable to find ProwJob'  but other errors are 'Failure locating ProwJob...

Reply:
Any error here is an error while running the search for a prow job. the first case is just one that's recognized (i.e., it isn't there, no point retrying). if something else goes wrong (connection to db lost or whatever) then that's passed through as a normal error.

Reply:
Yep, so you are -1 on changing the failure message :-)